core/shared/src/main/scala/com/monovore/decline/Help.scala

```diff
@@ -3,70 +3,130 @@ package com.monovore.decline
+      }
+
+    def intersperseList[A](xs: List[A], x: A): List[A] = {
+      val bld = List.newBuilder[A]
+      val it = xs.iterator
+      if (it.hasNext) {
+        bld += it.next
+        while (it.hasNext) {
+          bld += x
+          bld += it.next
         }
-        List(("Subcommands:" :: texts).mkString("\n"))
       }
+      bld.result
+    }
+
+    val optionsSection = {
+      val optionHelpLines =
+        intersperseList(
+          optionHelp.map(optHelp => withIndent(4, optHelp.show(theme))),
+          List("")
+        ).flatten
+
+      if (optionHelp.isEmpty) Nil
+      else (theme.sectionHeading("Options and flags:") :: optionHelpLines).mkString("\n") :: Nil
+    }
 
-    val optionsHelp = {
-      val optionsDetail = detail(parser.options)
-      if (optionsDetail.isEmpty) Nil
-      else ("Options and flags:" :: optionsDetail).mkString("\n") :: Nil
+    val envSection = {
+      if (envHelp.isEmpty) Nil
+      else
+        (theme.sectionHeading("Environment Variables:") :: envHelp
+          .flatMap(_.show(theme))
+          .map(withIndent(4, _)))
+          .mkString("\n") :: Nil
     }
 
-    val envVarHelp = {
-      val envVarHelpLines = environmentVarHelpLines(parser.options).distinct
-      if (envVarHelpLines.isEmpty) Nil
-      else ("Environment Variables:" :: envVarHelpLines.map("    " ++ _)).mkString("\n") :: Nil
+    val prefixString = prefix.mkString_(" ")
+
+    val usageSection = {
+      theme.sectionHeading("Usage:") :: usages.flatMap(us =>
+        us.show.map(line => withIndent(4, prefixString + " " + line))
+      )
     }
 
+    val errorsSection = if (args.errors.isEmpty) Nil else args.errors.map(theme.error(_))
+
+    val descriptionSection = List(description)
+
+    List(
+      errorsSection.mkString("\n"),
+      usageSection.mkString("\n"),
+      descriptionSection.mkString("\n"),
+      optionsSection.mkString("\n"),
+      envSection.mkString("\n"),
+      commandSection.mkString("\n")
+    ).filter(_.nonEmpty)
+      .mkString("\n\n")
+
+  }
+
+}
+
+object Help {
+
+  implicit val declineHelpShow: Show[Help] =
+    Show.fromToString[Help]
+
+  def fromCommand(parser: Command[_]): Help = {
     Help(
       errors = Nil,
-      prefix = NonEmptyList(parser.name, Nil),
-      usage = Usage.fromOpts(parser.options).flatMap { _.show },
-      body = parser.header :: (optionsHelp ::: envVarHelp ::: commandHelp)
+      prefix = NonEmptyList.of(parser.name),
+      usage = Nil,
+      body = Nil,
+      args = HelpArgs(
+        errors = Nil,
+        optionHelp = collectOptHelp(parser.options),
+        commandsHelp = collectCommandHelp(parser.options),
+        envHelp = collectEnvOptions(parser.options).distinct,
+        usages = Usage.fromOpts(parser.options),
+        description = parser.header
+      )
     )
+
   }
 
-  def optionList(opts: Opts[_]): Option[List[(Opt[_], Boolean)]] = opts match {
+  private[decline] case class HelpArgs(
```

Comment:
Neat! This seems like something with this shape could eventually be cleaned up and made public for some of the fancier help-rendering ideas folks have had (HTML generation, etc.) but leaving it private for now seems like the right call.